builtin/logical/database/backend.go

```diff
@@ -384,10 +336,10 @@ func (b *databaseBackend) ClearConnection(name string) error {
 // ClearConnectionId closes the database connection with a specific id and
 // removes it from the b.connections map.
 func (b *databaseBackend) ClearConnectionId(name, id string) error {
-	db := b.connPopIfEqual(name, id)
+	db := b.connections.PopIfEqual(name, id)
 	if db != nil {
 		// Ignore error here since the database client is always killed
-		db.Close()
+		(*db).Close()
```

Comment:
nit: I think this can stay as it was before now that the `PopIfEqual` signature has been updated.

```suggestion
		db.Close()
```